src/nupic/bindings/algorithms.i

```diff
@@ -299,10 +299,18 @@ void forceRetentionOfImageSensorLiteLibrary(void) {
     self->load(inStream);
   }
 
-  inline void add_sample(float y_val, PyObject* x_vector)
+  inline PyObject* add_sample(float y_val, PyObject* x_vector)
   {
-    PyArrayObject* x = (PyArrayObject*) x_vector;
+    PyArrayObject* x = (PyArrayObject*)x_vector;
+    if (!PyArray_ISFLOAT(x) || PyArray_ITEMSIZE(x) != sizeof(float))
```

Comment:
Why do we need `PyArray_ITEMSIZE(x) != sizeof(float)` in addition to `PyArray_ISFLOAT`? I expected `PyArray_ISFLOAT` to be sufficient.

Also,  the message below is `Expected x_vector dtype to be numpy.float32`, but the check `PyArray_ITEMSIZE(x) != sizeof(float)` seems to be against the plain C float, not "numpy.float32". What guarantee is there that `sizeof(float)` on a particular architecture matches size of "numpy.float32"?